### ISSUE TITLE
Bug: WSLg not support

### ISSUE THREAD

**Issue by LittleboyHarry** [RamonUnch/AltSnap#191]
https://user-images.githubusercontent.com/13063376/170675619-587155db-6226-4009-8a7f-16e765becace.mp4

Could you fix it? If not, maybe add exclude item `mstsc.exe` in:

https://github.com/RamonUnch/AltSnap/blob/077a09e58acd03d1cf6f3b6b9c0a54c02c38a135/AltSnap.ini#L243

**Comment by RamonUnch** [comment]
Yes `mstsc.exe` can be added to the blacklist, it would be useful for remote desktop connection, but it should not change anything on WSLg. By default the `*|TSSHELLWND` should be in the Windows blacklist for the same purpose (mstsc.exe main window).

I have no idea on how to move Windows inside WSLg. Probably you would need to write a program inside Linux environment or install an other window manager under your Linux Install.
I never setup a WSL environment And I no more have a Win10 computer and it runs slow as hell on any VM so I guess I will check this out when I get more time and a more modern PC.

Also do you get any details when using the **identify window** option in the blacklist tab on the config dialog?

**Comment by LittleboyHarry** [comment]
AutoHotKey, Windows Spy:
```
glxgears (arch)
ahk_class RAIL_WINDOW
ahk_exe mstsc.exe
ahk_pid 8504
```

Process Commandline:

    mstsc.exe /v:65D378AF-40D6-4FFB-88B9-7C157DD1161E /hvsocketserviceid:C02CFA8B-FACB-11E6-BD58-64006A7986D3 /silent /wslg /plugin:WSLDVC /wslgsharedmemorypath:WSL\65D378AF-40D6-4FFB-88B9-7C157DD1161E\wslg C:\ProgramData\Microsoft\WSL\wslg.rdp

**Comment by RamonUnch** [comment]
So it would seem that the Linux programs are embedded inside a `mstsc.exe` window, so in this case we need to be sure that mstsc windows are not in the blacklist.
Try just in case to remove the `*|TSSHELLWND` item in the `Windows` blacklist if it is present.
Also I would like if possible to see AltSnap own Identify Window that will tell us if AltSnap gets the same info than the ahk script.

**Comment by LittleboyHarry** [comment]
After tweak blacklist:

![0](https://user-images.githubusercontent.com/13063376/170849842-8a54e9ba-8c11-4c81-adfa-6b8d53a0a534.png)

One More Thing: Although the frontend visual window stuck, `glxgears` still log stable and normal render fps in the terminal background!

**Comment by RamonUnch** [comment]
Strange to see that it does not work, all seems to indicate that the window should be moveable. I really need to get Windows 11 at some point to make this kind of tests.
Maybe just in case, try to **elevae** AltSnap if it is not already the case. you can quickly try from the tray menu.

**Comment by RamonUnch** [comment]
Also seems to be related to https://github.com/microsoft/wslg/issues/574
So I guess older versions of AltSnap were working better. I am unsure however.

**Comment by sukerokushin** [comment]
still persists in the exact same way on wslg 1.0.41 and altsnap 1.54. the programs also stopped updating visually but were still working in the background, exactly as OP said.

**Comment by RamonUnch** [comment]
Well for now I have no access to Windows 11 and I cannot test WSLg.
At this moment you must add `mstsc.exe` to the process balcklist or add `*|RAIL_WINDOW` to the Windows blacklist.
